test/NuGet.Core.Tests/NuGet.Packaging.Test/ContentModelTests/PatternExpressionMatchTests.cs

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using NuGet.ContentModel;
+using NuGet.ContentModel.Infrastructure;
+using Xunit;
+
+namespace NuGet.Packaging.Test
+{
+    public class PatternExpressionMatchTests
+    {
+        [Fact]
+        public void Match_LiteralSegment_ShouldMatch()
+        {
+            // match literal segments given a pattern
+            Dictionary<string, ContentPropertyDefinition> propertyDefinitions = new()
+            {
+                { "type", new ContentPropertyDefinition("text") }
+            };
+            var path = "content/file.txt";
+            PatternDefinition pattern = new PatternDefinition(path);
+            PatternExpression expression = new PatternExpression(pattern);
+            ContentItem result = expression.Match(path, propertyDefinitions);
+            Assert.NotNull(result);
+            Assert.Equal(path, result.Path);
+        }
+
+        [Fact]
+        public void Match_LiteralSegment_ShouldNotMatch()
+        {
+            // should not match the literal segment pattern because file.txt != file2.txt
+            Dictionary<string, ContentPropertyDefinition> propertyDefinitions = new()
+            {
+                { "type", new ContentPropertyDefinition("text") }
+            };
+            PatternDefinition pattern = new PatternDefinition("content/file.txt");
+            PatternExpression expression = new PatternExpression(pattern);
+            var path = "content/file2.txt";
+            ContentItem result = expression.Match(path, propertyDefinitions);
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void Match_LiteralSegmentAndTokenSegment_ShouldMatch()
+        {
+            //We have a parser for the token "name" which takes in any type of string and returns it.
+            //Therefore file.txt should be added to the result content item
+            Dictionary<string, ContentPropertyDefinition> propertyDefinitions = new()
+            {
+                { "name", new ContentPropertyDefinition("any", parser: (o, t) => o) }
+            };
+            //this pattern has both a token segment and a literal segment
```

Comment:
NIT: Generally, an empty line should precede a line comment and succeed a closing bracket.